frontend/user/profileService.js

```diff
@@ -29,5 +44,55 @@
             return HttpService.patch(USER_URI, data);
         };
 
-    });
+        service._isAdmin = institution => user.isAdmin(institution.key);
+
+        service.removeProfile = (event, institution) => {
+            if (!service._isAdmin(institution)) {
+                const textContent = service._hasMoreThanOneInstitution() ? HAS_MORE_THAN_ONE_INSTITUTION_MSG : HAS_ONLY_ONE_INSTITUTION_MSG;
+                const confirm = $mdDialog.confirm();
+                confirm
+                    .clickOutsideToClose(false)
+                    .title('Remover vínculo com ' + institution.name)
+                    .textContent(textContent)
+                    .ariaLabel('Remover instituicao')
+                    .targetEvent(event)
+                    .ok('Sim')
+                    .cancel('Não');
+                const promise = $mdDialog.show(confirm);
+                promise.then(function () {
+                    service._deleteInstitution(institution.key);
+                }, function () {
+                    MessageService.showToast('Cancelado');
+                });
+                return promise;
+            } else {
+                MessageService.showToast('Desvínculo não permitido. Você é administrador dessa instituição.');
+            }
+        };
+
+        service._deleteInstitution = (institutionKey) => {
+            return new Promise(resolve => {
+                UserService.deleteInstitution(institutionKey)
```

Comment:
` UserService.deleteInstitution` is already a promise. you can return it directly